### PR TITLE
Added string literal docs

### PR DESCRIPTION
**Pull request by Qfl3x**
Issue #40825. Could use some suggestions for what to put.

### CODE REVIEW

**Review by stevengj — APPROVED** [review]

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -2698,4 +2698,42 @@ A quoted piece of code, that does not support interpolation. See the [manual sec
...
+"Contains \"quote\" characters"
+julia> \"""
+           Hello,
+           world.
+         \"""
+"  Hello,\n  world.\n"
+```
+See also ["](@ref \")
+"""
+kw"\"\"\""
```
Comment (simeonschaub) [inline base/docs/basedocs.jl:2737]: The code samples here don't look right.

**Review by simeonschaub — CHANGES_REQUESTED** [review]

File: base/docs/basedocs.jl (hunk 1 above)
Comment (Qfl3x) [reply base/docs/basedocs.jl]: Yeah, newline charcaters weren't escaped. I made a new commit that should fix the issue.

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -2698,4 +2698,42 @@ A quoted piece of code, that does not support interpolation. See the [manual sec
...
+"Contains \"quote\" characters"
+julia> \"""
+           Hello,
+           world.
+         \"""
+"  Hello,\\n  world.\\n"
+```
+See also ["](@ref \")
+"""
+kw"\"\"\""
```
Comment (simeonschaub) [inline base/docs/basedocs.jl:2737]: The formatting is still incorrect. I would recommend to always try the doctests out in the REPL first.
````suggestion
"""
    "
`"` Is used to delimit string literals.

# Examples

```jldoctest
julia> "Hello World!"
"Hello World!"

julia> "Hello World!\\n"
"Hello World!\\n"
```

See also [\"""](@ref \"\"\").
"""
kw"\""

"""
    \"""
`\"""` is used to delimit string literals. Strings created by triple quotation marks can contain " characters without escaping and are dedented to the level of the least-indented line. This is useful for defining strings within code that is indented.

# Examples

```jldoctest
julia> \"""Hello World!\"""
"Hello World!"

julia> \"""Contains "quote" characters\"""
"Contains \\"quote\\" characters"

julia> \"""
         Hello,
         world.\"""
"Hello,\\nworld."
```

See also ["](@ref \")
"""
kw"\"\"\""
````

Comment (Qfl3x) [reply base/docs/basedocs.jl]: I hope this last commit is it. There was also the newline in line 2748.

Comment (simeonschaub) [reply base/docs/basedocs.jl]:
````suggestion
"""
    "
`"` Is used to delimit string literals.

# Examples

```jldoctest
julia> "Hello World!"
"Hello World!"

julia> "Hello World!\\n"
"Hello World!\\n"
```

See also [`\"""`](@ref \"\"\").
"""
kw"\""

"""
    \"""
`\"""` is used to delimit string literals. Strings created by triple quotation marks can contain `"` characters without escaping and are dedented to the level of the least-indented line. This is useful for defining strings within code that is indented.

# Examples

```jldoctest
julia> \"""Hello World!\"""
"Hello World!"

julia> \"""Contains "quote" characters\"""
"Contains \\"quote\\" characters"

julia> \"""
         Hello,
         world.\"""
"Hello,\\nworld."
```

See also [`"`](@ref \")
"""
kw"\"\"\""
````

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -2698,4 +2698,42 @@ A quoted piece of code, that does not support interpolation. See the [manual sec
...
+"Contains \\"quote\\" characters"
+julia> \"""
+           Hello,
+           world.
+         \"""
+"  Hello,\\n  world.\\n"
+```
+See also ["](@ref \")
+"""
+kw"\"\"\""
```
Comment (simeonschaub) [inline base/docs/basedocs.jl:2737]: There are still some things missing in your commit. I think it's easier if I just commit what I have into your branch.
````suggestion
"""
    "
`"` Is used to delimit string literals.

# Examples

```jldoctest
julia> "Hello World!"
"Hello World!"

julia> "Hello World!\\n"
"Hello World!\\n"
```

See also [`\"""`](@ref \"\"\").
"""
kw"\""

"""
    \"""
`\"""` is used to delimit string literals. Strings created by triple quotation marks can contain `"` characters without escaping and are dedented to the level of the least-indented line. This is useful for defining strings within code that is indented.

# Examples

```jldoctest
julia> \"""Hello World!\"""
"Hello World!"

julia> \"""Contains "quote" characters\"""
"Contains \\"quote\\" characters"

julia> \"""
         Hello,
         world.\"""
"Hello,\\nworld."
```

See also [`"`](@ref \")
"""
kw"\"\"\""
````

**Comment by simeonschaub** [comment]
Thanks for the contribution!

**Pull request merged**